detection_rules/misc.py

```diff
@@ -290,10 +299,193 @@ def parse_config():
     return config
 
 
+def discover_tests(start_dir: str = 'tests', pattern: str = 'test*.py', top_level_dir: Optional[str] = None):
+    def list_tests(s, tests=None):
+        if tests is None:
+            tests = []
+        for test in s:
+            if isinstance(test, unittest.TestSuite):
+                list_tests(test, tests)
+            else:
+                tests.append(test.id())
+        return tests
+
+    loader = unittest.defaultTestLoader
+    suite = loader.discover(start_dir, pattern=pattern, top_level_dir=top_level_dir or str(ROOT_DIR))
+    return list_tests(suite)
+
+
+@dataclass
+class UnitTest:
+    bypass: Optional[List[str]] = None
+    test_only: Optional[List[str]] = None
+
+    def __post_init__(self):
+        assert not (self.bypass and self.test_only), 'Cannot use both test_only and bypass'
+
+
+@dataclass
+class RuleValidation:
+    bypass: Optional[List[str]] = None
+    test_only: Optional[List[str]] = None
+
+    def __post_init__(self):
+        assert not (self.bypass and self.test_only), 'Cannot use both test_only and bypass'
+
+
+@dataclass
+class TestConfig:
```

Comment:
I'd recommend moving this class to a dedicated module since it seems like it has a mode dedicated purpose than miscellaneous methods/classes (perhaps config.py).